### PR TITLE
Merging version v2.4.0 back to main for release

### PR DESCRIPTION
**Pull request by ed-xmos**
ADDED: Support for building the core ASRC code in the C emulator as a library
ADDED: Auto-generated ASRC and SSRC performance plots in documentation
CHANGED: Documents built under Jenkins instead of Github Actions
ADDED: Documentation warning about overflow in XS3 optimized SRC components
CHANGED: Tested against fwk_core v1.0.2 updated from v1.0.0

### CODE REVIEW

**Review by shuchitak — APPROVED** [review]
Looks good

File: doc/python/README.txt [hunk 1]
```diff
@@ -0,0 +1,37 @@
+INTRODUCTION
+============
+
+This folder includes a python script which runs the C emulators for the ASRC, SSRC, OS3 and DS3 modules.
+For each model we present test signals - both a single tone to test the THD performance and a range of tones to
+observe roll-off at the top of the band.
+Using an un-windowed FFT we calculate the SNR.
```
Comment (mbanth) [inline doc/python/README.txt:7]: Please avoid using personal pronouns in customer-visible documentation.

I have seen some use of 'we' in source file comments as well.  That's only a problem if they will be included in user-visible documentation via doxygen.

**Review by mbanth — APPROVED** [review]
One minor documentation comment which is optional to fix.  Otherwise, these changes look good.

File: doc/python/README.txt (hunk 1 above)
Comment (ed-xmos) [reply doc/python/README.txt:7]: I have updated this - I will not name the author.

**Pull request merged**